test/testautovariables.cpp

```diff
@@ -1183,6 +1188,161 @@ class TestAutoVariables : public TestFixture {
               "}");
     }
 
+    void danglingLifetimeLambda() {
+        check("auto f() {\n"
+              "    int a = 1;\n"
+              "    auto l = [&](){ return a; };\n"
+              "    return l;\n"
+              "}\n");
+        ASSERT_EQUALS("[test.cpp:2] -> [test.cpp:4]: (error) Returning object with dangling lifetime from local variable 'a'.\n", errout.str());
```

Comment:
I want that we test `ValueFlow` in testvalueflow. so most of these tests seem to belong in that file.

The algorithm of the checker is basically just to lookup "return|throw". And then if the operand has a dangling lifetime according to valueflow there is a error. We can test the "return|throw" lookup here and how the valueflow is used. But not the valueflow analysis.

In testvalueflow it should be tested if the operand get the value properly.

I am not sure how it should be splitted... but there is no test in testvalueflow right now.